packages/blade/src/components/Box/BaseBox/BaseBox.web.tsx

```diff
@@ -1,7 +1,13 @@
+import React from 'react';
 import styled from 'styled-components';
+import { getBaseBoxStyles, getDependencyProps } from './getBaseBoxStyles';
 import type { BaseBoxProps } from './types';
-import getBaseBoxStyles from './getBaseBoxStyles';
 
-const BaseBox = styled.div<BaseBoxProps>(getBaseBoxStyles);
+const BaseBox = styled.div<BaseBoxProps>((props) => {
+  const dependencyProp = getDependencyProps(props);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const cssObject = React.useMemo(() => getBaseBoxStyles(props), [dependencyProp]);
```

Comment:
Didn't quite get memoizing this based on `dependencyProp`. Is the intent to only recalculate the styles if one of the provided style prop is changed on `BaseBox`?

Reply:
Yup. Without memo, it runs that on all renders so it also calls `getBaseBoxStyles` when screen size is changing or sometimes randomly during scroll. Since we're just adding media queries, we don't want to change anything on screen size change.

Reply:
Can you perhaps add some comments in the code?

Reply:
donee